### PR TITLE
[#671] Invocation logic switch

### LINKED ISSUE #671: Model API invocation logic switch
We have to use HTTP URL arguments (`?a=v1&b=v2`) only in GET method, `application/x-www-form-urlencoded` data in POST requests. Raise exception in other cases (e.g. `form-data` POST mode with boundaries). Also we have to implement this logic switch in model API logging.

### PR DESCRIPTION
**Pull request by mcdoker18**
* Created the separate parse_invoke_request function because the
  parse_request function is used in other places
* Added unit tests
* Updated the rest api doc

This closes #671

### CODE REVIEW

**Comment by mcdoker18** [comment]
bot, test all

**Comment by mcdoker18** [comment]
bot, test all

**Comment by kirillmakhonin** [comment]
bot, test all

**Comment by mcdoker18** [comment]
bot, test all

**Comment by mcdoker18** [comment]
bot, test all

**Comment by alexander-semenets** [comment]
bot, test app

**Comment by mcdoker18** [comment]
bot, test app

File: legion/legion/http.py [hunk 1]
```diff
@@ -136,6 +136,25 @@ def parse_request(input_request):
     return result
 
 
+def parse_invoke_request(input_request):
```
[the remark this answers is not in the document]
Comment (dsuslov) [reply legion/legion/http.py]: @mcdoker18 parse_request() has to be replaced by parse_invoke_request() everywhere  parse_request() is used.

**Comment by mcdoker18** [comment]
bot, test all

**Comment by mcdoker18** [comment]
bot, test all

**Comment by mcdoker18** [comment]
bot, test all

**Pull request merged**